packages/tree/src/lib/JsonViewer.tsx

```diff
@@ -14,19 +14,21 @@ const isObject = (value: unknown): value is Record<string, unknown> => {
   return !!value && typeof value === 'object';
 };
 
-const getDetails = ({ data }: JsonViewerProps): (JSX.Element | null)[] => {
-  return Object.entries(data).map((entry) => {
+const getDetails = ({ data }: JsonViewerProps, parentKey?: string): (JSX.Element | null)[] => {
+  return Object.entries(data).map((entry, index) => {
     const [key, value] = entry;
 
     if (isPrimitive(value)) {
       const label = `${key}: ${value.toString()}`;
-      return <TreeItem label={label} itemId={`${key}.${value}`} key={`${key}.${value}`} />;
+      const id = parentKey ? `${parentKey}.${index}.${key}.${value}` : `${index}.${key}.${value}`;
```

Comment:
raw string keys and values will likely lead to invalid ids. Do we need all 3 (index, key, value) or can we just strip it down to one or two and validate strings?